pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -122,3 +127,36 @@ func retainJobSelectorFields(desired, observed *unstructured.Unstructured) (*uns
 	}
 	return desired, nil
 }
+
+const IndexKeyHPAScaleTargetRef = "spec.scaleTargetRef"
+
+func retainDeploymentFields(cli client.Client) retentionInterpreter {
+
+	return func(desired, observed *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+		hpaList := &autoscalingv2.HorizontalPodAutoscalerList{}
+		deployIdx := autoscalingv2.CrossVersionObjectReference{Kind: desired.GetKind(), Name: desired.GetName(), APIVersion: desired.GetAPIVersion()}
+
+		if err := cli.List(context.Background(), hpaList, client.InNamespace(desired.GetNamespace()),
```

Comment:
Have a question:

If we have another custom resources, How to implementation such similar logic?

Reply:
write a similar retain function？(Is custom resources somewhere more specifial in retain operator?)

besides, if you use `labels` method, where would you resolve custom resources retain operatoin? I think there's may be no major difference.

Reply:
>  if you use labels method, where would you resolve custom resources retain operatoin?

we can write a `lua` scripts, and we only use label


We can determine whether to execute the "retain" operation by writing a `Lua` script that `only focuses on` whether specific label exist in the resource's label.

If use your current approach, there is no `k8s client` in Lua.

Reply:
`k8s client` is not required.

> execute the "retain" operation by writing a Lua script

can this method use custom resources informer? As long as we can list the resources, we can implement similar logic.

Reply:
> can this method use custom resources informer?

To be honest, I don't know either.

Reply:
at this rate, labels method is better

Reply:
If we finally use the label method, can the current pr be closed first?

Reply:
yes